x-pack/plugins/data_enhanced/common/search/es_search/es_search_rxjs_utils.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { of, merge, timer, throwError } from 'rxjs';
+import { takeWhile, switchMap, expand, mergeMap } from 'rxjs/operators';
+
+import {
+  AbortError,
+  doSearch,
+  IKibanaSearchResponse,
+  isErrorResponse,
+} from '../../../../../../src/plugins/data/common';
+import type { IKibanaSearchRequest } from '../../../../../../src/plugins/data/common';
+import type { IAsyncSearchOptions } from '../../../common/search/types';
+
+const DEFAULT_POLLING_INTERVAL = 1000;
+
+export const doPartialSearch = <SearchResponse = any>(
```

Comment:
In general we should be extending from `IKibanaSearchRequest`/`IKibanaSearchResponse` and defaulting to `IEsSearchRequest`/`IEsSearchResponse`.

```suggestion
export const doPartialSearch = <SearchResponse extends IKibanaSearchResponse = IEsSearchResponse>(
```